### PR TITLE
Jest-based testing

### PR DESCRIPTION
**Pull request by hobinjk-ptc**
Replaces previous manual testing with jest-based integration tests. Notably adds a ton of logic to server's exit process to shut down existing intervals and close listening sockets instead of relying on process.exit to kill everything. Tests now include basic exercising of the object and frame APIs.

### CODE REVIEW

File: libraries/StaleObjectCleaner.js [hunk 1]
```diff
@@ -45,6 +47,13 @@ class StaleObjectCleaner {
         }
         return this.lastUpdates[objectKey];
     }
+
+    clearCleanupIntervals() {
+        for (let interval of this.cleanupIntervals) {
+            clearInterval(interval);
+        }
+        this.cleanupIntervals = [];
+    }
```
Comment (benptc) [inline libraries/StaleObjectCleaner.js:56]: Thanks for properly cleaning this up!

File: scripts/ci.sh [hunk 2]
```diff
@@ -9,8 +9,16 @@ git clone https://github.com/hobinjk-ptc/test-spatialToolbox spatialToolbox
 
 cd addons
 
+git clone https://github.com/ptcrealitylab/pop-up-onboarding-addon
```
Comment (benptc) [inline scripts/ci.sh:12]: I'm not sure exactly when this script gets executed, but just to double check – is it ok to point to this private addon repo in the open source server?

File: server.js [hunk 3]
```diff
@@ -1062,10 +1065,61 @@ function startSystem() {
...
-    process.exit();
+    await closeServer(http);
+    await closeServer(io.server.server);
+    sceneGraph.clearIntervals();
+    await recorder.stop();
+    clearActiveHeartbeats();
+    udpServer.close();
+    clearInterval(socketUpdaterInterval);
+    staleObjectCleaner.clearCleanupIntervals();
+    humanPoseFuser.stop();
```
Comment (benptc) [inline server.js:1119]: Nice updates to the exit process!

File: tests/create-frame.test.js [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+
+const frameMovedRef = JSON.parse(JSON.stringify(frameAddedRef));
+frameMovedRef.ar.matrix = [
+    0.966, 0.015, 0.257, 0,
+    -0.256, -0.0675, 0.964, 0,
+    -0.031, 0.997, 0.061, 0,
+    -150, -293, -555, 1
+];
+
+test('new object creation', async () => {
```
Comment (benptc) [inline tests/create-frame.test.js:156]: So nice to have these tests now!

**Review by benptc — APPROVED** [review]
Awesome updates!

File: scripts/ci.sh (hunk 2 above)
Comment (hobinjk-ptc) [reply scripts/ci.sh:12]: A good thing to double check! Any pull request made from an external repository won't have access to the secrets required to pull any of the private repositories. Additionally, any first-time contribution won't run the tests at all without our approval.

**Pull request merged**